### PR TITLE
feat(theming): add global style ability

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

Adding the ability to create global styles with the experimental theming APIs

```jsx
<GlobalStyle styles={{
  'body': {
    backgroundColor: 'purple'
    // supports design tokens!
    color: theme.tokens.colors.font.primary
  }
}} />
```

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/components/ThemeProvider/GlobalStyle.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  BaseComponentProps,
+  ElementType,
+  ForwardRefPrimitive,
+  Primitive,
+  PrimitiveProps,
+} from '../../primitives/types';
+import { primitiveWithForwardRef } from '../../primitives/utils/primitiveWithForwardRef';
+import { Style } from './Style';
+
+interface BaseGlobalStyleProps extends BaseComponentProps {
```
Comment (calebpollman) [inline packages/react/src/components/ThemeProvider/GlobalStyle.tsx:13]: Should this be extending `BaseComponentProps`?

File: packages/react/src/components/ThemeProvider/GlobalStyle.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  nonce?: string;
+  styles: Parameters<typeof createGlobalCSS>[0];
+}
+
+export type GlobalStyleProps<Element extends ElementType = 'style'> =
+  PrimitiveProps<BaseGlobalStyleProps, Element>;
+
+const GlobalStylePrimitive: Primitive<GlobalStyleProps, 'style'> = (
+  { styles, ...rest },
+  ref
```
Comment (calebpollman) [inline packages/react/src/components/ThemeProvider/GlobalStyle.tsx:28]: Does this component need a `ref`? Wondering if it complicates usage for RSC

File: packages/react/src/components/ThemeProvider/GlobalStyle.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { createGlobalCSS } from '@aws-amplify/ui';
+import { Style } from './Style';
+
+interface GlobalStyleProps extends React.StyleHTMLAttributes<{}> {
```
Comment (calebpollman) [inline packages/react/src/components/ThemeProvider/GlobalStyle.tsx:5]: Does `style` make sense as the element generic here?

```suggestion
interface GlobalStyleProps extends React.StyleHTMLAttributes<'style'> {
```

File: packages/react/src/components/ThemeProvider/ComponentStyle.tsx [hunk 4]
```diff
@@ -1,17 +1,9 @@
...
-  ForwardRefPrimitive,
-  Primitive,
-  PrimitiveProps,
-} from '../../primitives/types';
-import { primitiveWithForwardRef } from '../../primitives/utils/primitiveWithForwardRef';
 import { BaseComponentTheme } from '@aws-amplify/ui';
 import { Style } from './Style';
 
-interface BaseComponentStyleProps extends BaseComponentProps {
+interface ComponentStyleProps extends React.StyleHTMLAttributes<{}> {
```
Comment (calebpollman) [inline packages/react/src/components/ThemeProvider/ComponentStyle.tsx:6]: Same thought as [here](https://github.com/aws-amplify/amplify-ui/pull/5825/files#diff-43aa10f06463990535af191d9d2710c6b14c2cc55810905f4bc5444cf9c6dde2R5)

```suggestion
interface ComponentStyleProps extends React.StyleHTMLAttributes<'style'> {
```

**Review by calebpollman — APPROVED** [review]
🚢

**Review by reesscot — DISMISSED** [review]

**Review by reesscot — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**